src/main/java/org/javarosa/xform/parse/FormInstanceParser.java

```diff
@@ -348,7 +348,7 @@ private void verifyItemsetBindings(FormInstance instance) {
 
             // Don't try to validate references if the external instance could not be resolved. We allow parsing a form
             // with placeholder external secondary instances for cases where a ReferenceManager can't be configured.
-            if (fi instanceof ExternalDataInstance && !((ExternalDataInstance) fi).isUsingPlaceholder()) {
+            if (!(fi instanceof ExternalDataInstance) || !((ExternalDataInstance) fi).isUsingPlaceholder()) {
```

Comment:
While we're here, let's change that `fi` variable to something more readable. It's especially mad here where we end up with and `if` in front of it.